hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -322,9 +328,10 @@ private OmMetadataManagerImpl(OzoneConfiguration conf, String snapshotDirName)
    * @throws IOException
    */
   public static OmMetadataManagerImpl createSnapshotMetadataManager(
-      OzoneConfiguration conf, String snapshotDirName) throws IOException {
-    OmMetadataManagerImpl smm = new OmMetadataManagerImpl(conf,
-        snapshotDirName);
+      OzoneConfiguration conf, String snapshotDirName,
+      boolean isSnapshotInCache) throws IOException {
+    OmMetadataManagerImpl smm = new OmMetadataManagerImpl(conf, snapshotDirName,
```

Comment:
nit: could be a one line.
```suggestion:
    return new OmMetadataManagerImpl(conf, snapshotDirName,
        isSnapshotInCache);
```

Also why this needs to be a static function. Why can't we use constructor directly? It is misuse of factory.

Reply:
Done. removed the static method and used constructor directly